cmd/object-handlers.go

```diff
@@ -1786,6 +1786,18 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 	}
 	opts.IndexCB = idxCb
 
+	if !opts.MTime.IsZero() && opts.PreserveETag != "" {
```

Comment:
These checks are redundant since we check for these in `checkPreconditionsPUT`.

Reply:
We do not want these checks to be initialized since they will call an additional read of xl.meta for normal operations @krisis